modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java

```diff
@@ -1221,6 +1225,56 @@ private void _removeFieldsFromDataListViews(
 		}
 	}
 
+	private void _reorderNestedDDMFormFields(DDMForm ddmForm) throws Exception {
+		for (DDMFormField ddmFormField : ddmForm.getDDMFormFields()) {
+			if (!StringUtil.equals(
+					ddmFormField.getType(),
+					DDMFormFieldTypeConstants.FIELDSET)) {
+
+				continue;
+			}
+
+			Map<String, DDMFormField> nestedDDMFormFieldsMap =
+				ddmFormField.getNestedDDMFormFieldsMap();
```

Comment:
@marcelabc I would move this to just before the `for (int i = 0; i < rowsJSONArray.length(); i++) {`, since we're only using this variable inside the `for`

Reply:
Don't you think it would be better to maintain the way it is? Because, If I move to this inside this for (rows), this variable would be created many times

Reply:
I meant the following:

```
Map<String, DDMFormField> nestedDDMFormFieldsMap = ddmFormField.getNestedDDMFormFieldsMap();

for (int i = 0; i < rowsJSONArray.length(); i++) { 
```

But I see no problem keeping it the way it is.

Reply:
Oh, now I got it! Thanks